rpc/client_test.go

```diff
@@ -51,6 +51,38 @@ func TestClientRequest(t *testing.T) {
 	}
 }
 
+func TestClientNamedParams(t *testing.T) {
+	client := Client{}
+
+	type params struct {
+		Str  string
+		Int  int
+		Args *echoArgs
+	}
+
+	p := params{
+		Str:  "hello",
+		Int:  10,
+		Args: &echoArgs{"world"},
+	}
+
+	method := "test_echo"
+	msg, err := client.newMessage(method, NewNamedParams(p))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if msg.Method != method {
+		t.Fatalf("expect method %s but got %s", method, msg.Method)
+	}
+	var msgParams params
+	if err := json.Unmarshal(msg.Params, &msgParams); err != nil {
+		t.Fatal(err)
+	}
+	if !reflect.DeepEqual(msgParams, p) {
+		t.Errorf("incorrect params %#v", msgParams)
+	}
+}
+
```

Comment:
The goal of this test is to ensure that request params are correctly encoded when `NamedParams` are passed as the sole argument. Testing named params with an actual server would require that the server supports named params, which I don't think this package's server implementation does. Changing that would be beyond the scope of this PR since the goal here is just to allow the client to be able to communicate with servers that do support named params.

Reply:
Yes, I understand.